src/output_writers.jl

```diff
@@ -224,25 +247,55 @@ end
 savesubstructs!(file, model, names) = [savesubstruct!(file, model, name) for name in names]
 
 function write_output(model, fw::JLD2OutputWriter)
-    @info @sprintf("Calculating JLD2 output %s...", keys(fw.outputs))
-    @time data = Dict((name, f(model)) for (name, f) in fw.outputs)
+    verbose = fw.verbose
+    verbose && @info @sprintf("Calculating JLD2 output %s...", keys(fw.outputs))
+    tc = Base.@elapsed data = Dict((name, f(model)) for (name, f) in fw.outputs)
+    verbose && @info "Calculation time: $(prettytime(tc))"
 
     iter = model.clock.iteration
     time = model.clock.time
+
+    filesize(fw.filepath) >= fw.max_filesize && start_next_file(model, fw)
+
     path = fw.filepath
+    verbose && @info "Writing JLD2 output $(keys(fw.outputs))..."
+    t0, sz = time_ns(), filesize(path)
 
-    @info @sprintf("Writing JLD2 output %s...", keys(fw.outputs))
-    t0 = time_ns()
-    if fw.asynchronous
+    if fw.async
         @async remotecall(jld2output!, 2, path, iter, time, data)
     else
         jld2output!(path, iter, time, data)
     end
-    @info "Done writing (t: $(prettytime(time_ns()-t0)))"
+
+    t1, newsz = time_ns(), filesize(path)
+    verbose && @info "Writing done: time=$(prettytime((t1-t0)/1e9)), size=$(pretty_filesize(newsz)), Δsize=$(pretty_filesize(newsz-sz))"
 
     return nothing
 end
 
+function start_next_file(model::Model, fw::JLD2OutputWriter)
+    verbose = fw.verbose
+    sz = filesize(fw.filepath)
+    verbose && @info "Filesize $(pretty_filesize(sz)) has exceeded maximum file size $(pretty_filesize(fw.max_filesize))."
+
+    if fw.part == 1
+        part1_path = replace(fw.filepath, r".jld2$" => "_part1.jld2")
+        verbose && @info "Renaming first part: $(fw.filepath) -> $part1_path"
+        mv(fw.filepath, part1_path, force=fw.force)
+        fw.filepath = part1_path
+    end
+
+    fw.part += 1
+    fw.filepath = replace(fw.filepath, r"part\d+.jld2$" => "part" * string(fw.part) * ".jld2")
```

Comment:
I think it'd be best to write a function that defines this filename convention for clarity, and to make it easier to change the convention if that is necessary.

Reply:
Yes splitting it up would make the code clearer. Will do.